[BACKLOG-5343]  Data Service ResultSet implementation mishandles getXXX(String)

mkambol opened the pull request:
Formerly shifted index by 1, since the RowMeta.indexOfValue is 0 based
and the .getXXX( int ) assumes 1 based.

@bmorrise @hudak

hudak commented on pdi-dataservice-client/src/main/java/org/pentaho/di/trans/dataservice/jdbc/ThinResultSet.java:
```diff
@@ -790,7 +793,7 @@ public String getString( int index ) throws SQLException {
   @Override
   public String getString( String columnName ) throws SQLException {
-    return getString( rowMeta.indexOfValue( columnName ) );
+    return getString( rowMeta.indexOfValue( columnName ) + 1 );
```
If a column name isn't found on the row, getXXX( 0 ) will be called, which means currentRow[ -1 ] will be accessed
Another IndexOutOfBounds will be thrown. Is that okay?

mkambol committed:
[BACKLOG-5343]  Data Service ResultSet implementation mishandles
getXXX(String)

Formerly shifted index by 1, since the RowMeta.indexOfValue is 0 based
and the .getXXX( int ) assumes 1 based.
Also adds a check that the column is present, throwing a SQLException
otherwise.  A SQLException was thrown before, but wrapped a
misleading IIOB.

dkincade commented:
devci: http://devci.pentaho.com/job/6.0-snapshot/job/pdi-dataservice-plugin/164/
